diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 9d51f99356..d6bdb57e08 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -60,7 +60,7 @@
  * 32M uncompressed kernel, 16M compressed kernel, 1M fdt,
  * 1M script, 1M pxe, 1M dt overlay and the ramdisk at the end.
  */
-#define BOOTM_SIZE        __stringify(0xa000000)
+#define BOOTM_SIZE        __stringify(0xe000000)
 #define KERNEL_ADDR_R     __stringify(SDRAM_OFFSET(2000000))
 #define FDT_ADDR_R        __stringify(SDRAM_OFFSET(3000000))
 #define SCRIPT_ADDR_R     __stringify(SDRAM_OFFSET(3100000))