diff --git a/common/bootm.c b/common/bootm.c
index 95cde11827e..9409aba5bd9 100644
--- a/common/bootm.c
+++ b/common/bootm.c
@@ -37,27 +37,6 @@
 
 #ifndef USE_HOSTCC
 
-#ifdef CONFIG_SUNXI_COMP_DETECT
-/*
- * Compression type and magic number mapping table.
- */
-struct comp_magic_map {
-	int		comp_id;
-	const char	*name;
-	unsigned char	magic[2];
-};
-
-static const struct comp_magic_map image_comp[] = {
-	{	IH_COMP_BZIP2,	"bzip2",	{0x42, 0x5a},},
-	{	IH_COMP_GZIP,	"gzip",		{0x1f, 0x8b},},
-	{	IH_COMP_LZMA,	"lzma",		{0x5d, 0x00},},
-	{	IH_COMP_LZO,	"lzo",		{0x89, 0x4c},},
-	{	IH_COMP_LZ4,    "lz4",          {0x04, 0x22},},
-	//{	IH_COMP_ZSTD,   "zstd",         {0x28, 0xb5},},
-	{	IH_COMP_NONE,	"none",		{},	},
-};
-#endif
-
 DECLARE_GLOBAL_DATA_PTR;
 
 bootm_headers_t images;		/* pointers to os/initrd/fdt images */
@@ -449,23 +428,6 @@ static int handle_decomp_error(int comp_type, size_t uncomp_size, int ret)
 
 	return BOOTM_ERR_RESET;
 }
-
-#ifdef CONFIG_SUNXI_COMP_DETECT
-int image_decomp_type(const unsigned char *buf, ulong len)
-{
-	const struct comp_magic_map *cmagic = image_comp;
-
-	if (len < 2)
-		return -EINVAL;
-
-	for (; cmagic->comp_id > 0; cmagic++) {
-		if (!memcmp(buf, cmagic->magic, 2))
-			break;
-	}
-
-	return cmagic->comp_id;
-}
-#endif
 #endif
 
 #ifndef USE_HOSTCC
